conn.go

```diff
@@ -276,9 +293,11 @@ func (l *Conn) finishMessage(messageID int64) {
 	}
 	l.messageMutex.Unlock()
 
+	close(msgCtx.done)
```

Comment:
shouldn't this happen before the the `isClosing` check? otherwise we may still have potential for deadlocks